Composer/packages/lib/code-editor/src/lu/dialogs/ListEntityCreationDialog.tsx

```diff
@@ -0,0 +1,308 @@
+            idx === foundIdx ? { ...foundItem, synonyms } : item
+          );
+        }
+
+        return clonedEntity;
+      });
+    },
+    []
+  );
+
+  const deleteSelectedListEntityItems = React.useCallback(() => {
+    setListEntity((currentEntity) => {
+      const clonedEntity = { ...currentEntity };
+      clonedEntity.items = clonedEntity.items.filter((item) => !selectedItems.map((si) => si.id).includes(item.id));
+
+      return clonedEntity;
+    });
+  }, [selectedItems]);
+
+  const scrollToItemByIndex = React.useCallback((idx: number) => {
+    scrollTimeoutRef.current = setTimeout(() => {
+      listRootRef.current
+        ?.querySelector(`div.ms-List-cell[data-list-index="${idx}"]`)
+        ?.scrollIntoView({ behavior: 'smooth' });
+      scrollTimeoutRef.current = undefined;
+    }, 300);
+  }, []);
+
+  const addListEntityItem = React.useCallback(() => {
+    setListEntity((currentEntity) => {
+      const clonedEntity = { ...currentEntity };
+      clonedEntity.items = [...clonedEntity.items, { id: listEntityId, normalizedValue: '', synonyms: [] }];
+      listEntityId++;
+
+      scrollToItemByIndex(currentEntity.items.length);
+      return clonedEntity;
+    });
+  }, []);
+
+  const commandBarItems = React.useMemo<ICommandBarItemProps[]>(
+    () => [
+      {
+        key: 'addItem',
+        iconProps: { iconName: 'Add' },
+        name: formatMessage('Add row'),
+        onClick: addListEntityItem,
+      },
+      {
+        key: 'deleteItem',
+        iconProps: { iconName: 'Trash' },
+        name: formatMessage('Delete'),
+        disabled: !selectedItems.length,
+        onClick: deleteSelectedListEntityItems,
+      },
+    ],
+    [deleteSelectedListEntityItems, addListEntityItem, selectedItems]
+  );
+
+  const columns = React.useMemo<IColumn[]>(() => {
+    return [
+      {
+        key: 'normalizedValue',
+        name: formatMessage('Normalized value'),
+        isResizable: true,
+        onRender: (item: ListEntityItem) => {
+          return (
+            <TextField
+              borderless
+              data-selection-disabled
+              autoComplete="off"
+              errorMessage={itemsTouched[item.id] ? itemErrors[item.id] : ''}
+              placeholder={formatMessage('Enter a value')}
+              styles={normalizedValueTextField}
+              value={item.normalizedValue}
+              onChange={changeNormalizedValue(item)}
+            />
+          );
+        },
+        minWidth: 200,
+        maxWidth: 250,
+      },
+      {
+        key: 'synonyms',
+        name: formatMessage('Synonyms'),
+        isResizable: true,
+        onRender: (item: ListEntityItem) => {
+          const values = item.synonyms;
+          return (
+            <SynonymInput
+              removeOnBackspace
+              editable={false}
+              placeholder={formatMessage('Enter a synonym and press enter')}
+              tags={values}
```

Comment:
No need for single use variable
```suggestion
              tags={item.synonyms}
```